### PR TITLE
Misc windows fixes

### PR DESCRIPTION
**Pull request by tamird**
Decided to update to 5.1.2 while I'm here, but let me know if you'd prefer that as a separate change.

### CODE REVIEW

File: import.sh [hunk 1]
```diff
@@ -30,28 +30,29 @@
...
 rm -rf internal/*
-find . -type l -not -path './.git/*' | xargs rm
-curl -sL https://github.com/facebook/rocksdb/archive/v5.0.2.tar.gz | tar zxf - -C internal --strip-components=1
+find . -type l -not -path './.git/*' -exec rm {} \;
+curl -sL https://github.com/facebook/rocksdb/archive/v5.1.2.tar.gz | tar zxf - -C internal --strip-components=1
 make -C internal util/build_version.cc
-# TODO(tamird,petermattis): remove when
-# https://github.com/facebook/rocksdb/commit/4a17b47bb5a2cc09f95acb53ec565bccd171ee4e
-# is released.
-patch -p1 < testharness.patch
```
Comment (bdarnell) [inline import.sh]: Delete the file testharness.patch

File: import.sh [hunk 2]
```diff
@@ -30,28 +30,29 @@
...
+# TODO(tamird): remove when
+# https://github.com/facebook/rocksdb/pull/1910 is merged and release.
+patch -p1 < gettimeofday.patch
 # Downcase some windows-only includes for compatibility with mingw64.
-grep -lRF '<Windows.h>' internal | xargs sed -i~ 's!<Windows.h>!<windows.h>!g'
-grep -lRF '<Rpc.h>' internal | xargs sed -i~ 's!<Rpc.h>!<rpc.h>!g'
+grep -lR '#include <.*[A-Z].*>' internal | while IFS= read -r source_file; do
+	grep -o '#include <.*[A-Z].*>' "$source_file" | while IFS= read -r original; do
+		downcased=$(echo "$original" | tr '[:upper:]' '[:lower:]')
+		sed -i~ "s!$original!$downcased!g" "$source_file"
```
Comment (bdarnell) [inline import.sh]: This is much easier in `awk`: `awk '/#include/ { print tolower($0); next; } { print; }'`. Unfortunately `awk` doesn't have the handy `-i` option unless we're willing to rely on gnu extensions (and speaking of which, gnu sed has an extension for upper- and lower-casing).

**Review by bdarnell — APPROVED** [review]
I'm OK with moving ahead with 5.1.2 but @petermattis should weigh in too.

File: import.sh (hunk 2 above)
Comment (tamird) [reply import.sh]: That's not quite equivalent (it produced this diff):
```diff
diff --git a/internal/port/win/env_win.cc b/internal/port/win/env_win.cc
index 61d5cf1..df4a57c 100644
--- a/internal/port/win/env_win.cc
+++ b/internal/port/win/env_win.cc
@@ -32,7 +32,7 @@
 #include "util/thread_status_updater.h"
 #include "util/thread_status_util.h"
 
-#include <rpc.h>  // For UUID generation
+#include <rpc.h>  // for uuid generation
 #include <windows.h>
 
 namespace rocksdb {

```

But done anyway.

File: import.sh (hunk 1 above)
Comment (tamird) [reply import.sh]: It's already deleted here.

**Review by petermattis — COMMENTED** [review]
I'd prefer a separate change for the 5.1.2 bump, though I do want to do that.

**Comment by tamird** [comment]
Done.

**Pull request merged**